repos/spack_repo/builtin/packages/py_cupy/package.py

```diff
@@ -39,38 +43,52 @@ class PyCupy(PythonPackage, CudaPackage, ROCmPackage):
     depends_on("cxx", type="build")
 
     depends_on("python@3.7:", when="@:11", type=("build", "run"))
-    depends_on("python@3.8:", when="@12:", type=("build", "run"))
+    depends_on("python@3.8:3.11", when="@12", type=("build", "run"))
+    depends_on("python@3.9:3.13", when="@13", type=("build", "run"))
+    depends_on("python@3.10:", when="@14", type=("build", "run"))
     depends_on("py-setuptools", type="build")
     depends_on("py-cython@0.29.22:2", type="build", when="@:13.3")
     depends_on(
         "py-cython@3:3.0.10,3.0.12:", type="build", when="@13.4:"
     )  # 3.0.11 broken likely because of cython#6335, fixed in 3.0.12
     depends_on("py-fastrlock@0.5:", type=("build", "run"))
     depends_on("py-numpy@1.20:1.25", when="@:11", type=("build", "run"))
-    depends_on("py-numpy@1.20:1.26", when="@12:", type=("build", "run"))
-    depends_on("py-numpy@1.22:1.28", when="@13:", type=("build", "run"))
-
-    depends_on("py-scipy@1.6:1.12", when="@:12+all", type=("build", "run"))
-    depends_on("py-scipy@1.7:1.13", when="@13:+all", type=("build", "run"))
-    depends_on("py-cython@0.29.22:2", when="+all", type=("build", "run"))
+    depends_on("py-numpy@1.20:1.26", when="@12", type=("build", "run"))
+    depends_on("py-numpy@1.22:1", when="@13.1", type=("build", "run"))
+    depends_on("py-numpy@1.22:2.0", when="@13.2", type=("build", "run"))
+    depends_on("py-numpy@1.22:2.2", when="@13.4", type=("build", "run"))
+    depends_on("py-numpy@1.22:2.3", when="@13.5", type=("build", "run"))
+    depends_on("py-numpy@1.24:2", when="@14", type=("build", "run"))
+    depends_on("py-scipy@1.6:1.11", when="@:12+all", type=("build", "run"))
+    depends_on("py-scipy@1.7:1.12", when="@13+all", type=("build", "run"))
```

Comment:
From `pyproject.toml` for the version (https://github.com/cupy/cupy/blob/v13.5.1/pyproject.toml), this should be:

```suggestion
    depends_on("py-scipy@1.7:1.16", when="@13+all", type=("build", "run"))
```

Add a comment if that's not the case.

Reply:
You are right, I misadjusted here.